src/adagrams.js

```diff
@@ -1,15 +1,153 @@
+  console.log(letter);
+  for (let i = 0; i < letterPool[letter]; i++) {
+    lettersPool.push(letter);
+  }
+});
+console.log(lettersPool);
+
 export const drawLetters = () => {
-  // Implement this method for wave 1
+  let hand = [];
+  let letterPoolCopy = [...lettersPool];
+  for (let i = 0; i < 10; i++) {
+    let randomIndex = Math.floor(
+      Math.random() * Object.keys(letterPool).length
+    );
+    // Create a random letter from the list of letters
+    // using the random index
+    let selectedLetter = letterPoolCopy[randomIndex];
+    hand.push(selectedLetter);
+    letterPoolCopy.splice(randomIndex, 1);
+  }
+  return hand;
 };
 
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+  // create variable to count occurrences of each letter in word
+  input = input.toUpperCase();
+  let input_letter_count = {};
+  for (let i = 0; i < input.length; i++) {
+    let letter = input[i];
+    input_letter_count[letter] = (input_letter_count[letter] || 0) + 1;
+  }
+
+  // create variable to count occurrences of each letter in the letter bank
+  let bank_letter_count = {};
+  for (let i = 0; i < lettersInHand.length; i++) {
+    let letter = lettersInHand[i];
+    bank_letter_count[letter] = (bank_letter_count[letter] || 0) + 1;
+  }
+
+  // compare the input_letter_count and bank_letter_count to see if input can be made
+  for (let letter in input_letter_count) {
+    let count = input_letter_count[letter];
+    // check if not in bank_letter_count
+    if (!(letter in bank_letter_count) || count > bank_letter_count[letter]) {
+      return false;
+    }
+  }
+
+  return true;
 };
 
 export const scoreWord = (word) => {
-  // Implement this method for wave 3
+  let letter_values = {
+    A: 1,
+    B: 3,
+    C: 3,
+    D: 2,
+    E: 1,
+    F: 4,
+    G: 2,
+    H: 4,
+    I: 1,
+    J: 8,
+    K: 5,
+    L: 1,
+    M: 3,
+    N: 1,
+    O: 1,
+    P: 3,
+    Q: 10,
+    R: 1,
+    S: 1,
+    T: 1,
+    U: 1,
+    V: 4,
+    W: 4,
+    X: 8,
+    Y: 4,
+    Z: 10,
+  };
+
+  let score = 0;
+  word = word.toUpperCase();
+
+  for (let i = 0; i < word.length; i++) {
+    let letter = word[i];
+    score += letter_values[letter] || 0;
+  }
+
+  if (word.length >= 7 && word.length <= 10) {
+    score += 8;
+  }
+
+  return score;
 };
 
 export const highestScoreFrom = (words) => {
-  // Implement this method for wave 4
+  let highestScore = 0;
+  let winningWord = '';
+
+  // check through each
+  // console.log(108, words);
+  for (let i = 0; i < words.length; i++) {
+    const word = words[i];
+    const score = scoreWord(word);
+
+    if (score > highestScore) {
+      highestScore = score;
+      winningWord = word;
+    } else if (score === highestScore) {
+      if (
+        winningWord.length !== 10 &&
+        (word.length === 10 || word.length < winningWord.length)
+      ) {
+        winningWord = word;
+      }
+    }
+  }
+
+  return { word: winningWord, score: highestScore };
```

Comment:
Very easy to follow!

Reply:
Nice job, Raina! As you probably noticed, I didn't give too much feedback due to the scope of the project (Translating your python code to Javascript code). I tried to just point out any glaring issues, if any. With that being said, if you want to discuss anything in greater detail, feel free to reach out to me!